soroban-env-host/src/test/map.rs

```diff
@@ -372,3 +369,182 @@ fn large_map_exceeds_budget() {
         (Instant::now() - start).as_secs_f64()
     );
 }
+
+#[test]
+fn initialization_invalid() -> Result<(), HostError> {
+    use crate::EnvBase;
+    let host = observe_host!(Host::default());
+
+    // Out of order keys
+    assert!(host
+        .map_new_from_slices(&["b", "a"], &[1u32.into(), 2u32.into()])
+        .is_err());
+
+    // Duplicate
+    assert!(host
+        .map_new_from_slices(&["a", "a"], &[1u32.into(), 2u32.into()])
+        .is_err());
+
+    // large map from slices
+    let buf = vec![0; 7_000_000];
+    let scv_bytes = ScVal::Bytes(buf.try_into().unwrap());
+    let bytes_val = host.to_host_val(&scv_bytes)?;
+
+    let vals = vec![bytes_val; 5];
+    let keys = ["a", "b", "c", "d", "e"];
+
+    let map = host.map_new_from_slices(&keys, &vals.as_slice())?;
+    let res = host.from_host_val(map.to_val());
+    // This is actually Budget::ExceededLimit, but ScVal::try_from_val converts
+    // he error to a ConversionError. We have an issue for it https://github.com/stellar/rs-soroban-env/issues/1046
```

Comment:
We probably should triage this issue and either fix it now, or close it. I think it's a bit better to have a budget error, but if the fix takes more than updating a couple lines of code, then it's probably not worth it.

Reply:
The fix is unfortunately quite complex as it involves iterating on common and sdk code and macro code to find a stable trait arrangement. I tried once and was defeated. It's one of the things I decided we just don't have time or effort available for, but I feel bad about it.

Reply:
(I've pinged @leighmcculloch in case he has time for it, I really wanted to find time before final to fix it but at this point I expect we're going to be stuck with this behaviour forever)